build.sbt

```diff
@@ -181,11 +181,51 @@ addCommandAlias(
   "+publishLocal; metals/runMain scala.meta.metals.DownloadDependencies "
 )
 
-commands += Command.command("save-expect") { s =>
-  "unit/test:runMain tests.SaveExpect" ::
-    s
+def configureMtagsScalaVersionDynamically(
+    state: State,
+    scalaV: String,
+    projectV: Option[String] = None
+): State = {
+  val scalaVersionSettings =
+    List(
+      mtest / scalaVersion := scalaV,
+      mtags / scalaVersion := scalaV,
+      cross / scalaVersion := scalaV
+    )
+  val versionSettings =
+    projectV.map(v => mtags / version := v)
+
+  val extracted = Project.extract(state)
+  extracted
+    .appendWithSession(
+      scalaVersionSettings ++ versionSettings,
+      state
+    )
+}
+
+def crossTestDyn(state: State, scalaV: String): State = {
+  val configured = configureMtagsScalaVersionDynamically(state, scalaV)
+  val (out, _) =
+    Project
+      .extract(configured)
+      .runTask(cross / Test / test, configured)
+  out
 }
 
+commands ++= Seq(
+  Command.command("save-expect") { s =>
+    "unit/test:runMain tests.SaveExpect" :: s
+  },
+  Command.command("cross-test-latest-nightly") { s =>
+    V.nightlyScala3Versions.lastOption match {
```

Comment:
Would we want to test it for all last 5 versions?

Reply:
I think having latest should be enough. Will save some minutes per ci run.